internal/provider/zip_archiver_test.go

```diff
@@ -167,3 +191,27 @@ func ensureContent(t *testing.T, wants map[string][]byte, got *zip.File) {
 		t.Errorf("mismatched content\ngot\n%s\nwant\n%s", gotContent, wantContent)
 	}
 }
+
+func ensureFileMode(t *testing.T, zipfilepath string, outputFileMode string) {
+	r, err := zip.OpenReader(zipfilepath)
```

Comment:
Please add `t.Helper()` to `ensureFileMode` (and to `ensureContent` and `ensureContents` while you're at it, if you don't mind).